src/core/utilities.js

```diff
@@ -67,23 +67,54 @@ export function inArray( elem, array ) {
 }
 
 /**
- * Makes a clone of an object using only Array or Object as base,
- * and copies over the own enumerable properties.
+ * Recursively clone an object into a plain array or object, taking only the
+ * own enumerable properties.
  *
- * @param {Object} obj
- * @return {Object} New object with only the own properties (recursively).
+ * @param {any} obj
+ * @param {bool} [allowArray=true]
+ * @return {Object|Array}
  */
-export function objectValues( obj ) {
-	const vals = is( "array", obj ) ? [] : {};
+export function objectValues( obj, allowArray = true ) {
+	const vals = ( allowArray && is( "array", obj ) ) ? [] : {};
 	for ( const key in obj ) {
 		if ( hasOwn.call( obj, key ) ) {
 			const val = obj[ key ];
-			vals[ key ] = val === Object( val ) ? objectValues( val ) : val;
+			vals[ key ] = val === Object( val ) ? objectValues( val, allowArray ) : val;
 		}
 	}
 	return vals;
 }
 
+/**
+ * Recursively clone an object into a plain object, taking only the
+ * subset of own enumerable properties that exist a given model.
+ *
+ * @param {any} obj
+ * @param {any} model
+ * @return {Object}
+ */
+export function objectValuesSubset( obj, model ) {
+
+	// Return primitive values unchanged to avoid false positives or confusing
+	// results from assert.propContains().
+	// E.g. an actual null or false wrongly equaling an empty object,
+	// or an actual string being reported as object not matching a partial object.
+	if ( obj !== Object( obj ) ) {
+		return obj;
+	}
+
+	// Unlike objectValues(), subset arrays to a plain objects as well.
+	// This enables subsetting [20, 30] with {1: 30}.
+	const subset = {};
+
+	for ( const key in model ) {
+		if ( hasOwn.call( model, key ) && hasOwn.call( obj, key ) ) {
```

Comment:
Considering we already iterate on model keys with `const key in model`, can this line become the one below?

```
if (hasOwn.call( obj, key ) ) {
 // ...
}
```

> // This enables subsetting [20, 30] with {1: 30}.

I prefer if we don't have this feature actually :) I think this could be a feature that we don't ship, if we dont need to, considering:
- single values would be checked with [index] reference and equal/notEqual or deepEqual.
- subset of an array can be changed from beginning or end with `Array.prototype.slice`. For majority of the users, I suspect they would like to always check a complex/nested object in separate assertion in my view, thats probably 99% of the use cases, wdyt?
- I think this is less readable/maintainable for users: `assert.propContains(pageModelsArray, { 0: izel, 3: timo });` and might be making the test runtime more complex(with extra internal hasOwn checks) and slower(due to more if checks for smt that could be done in multiple assertions faster). 

If we dont support array subsetting, could this line also be removed? If we cant remove it, could it be reduced to single hasOwnProperty check when we don't support array subsetting?